newIDE/app/src/ObjectsRendering/PixiResourcesLoader.js

```diff
@@ -124,6 +131,55 @@ export default class PixiResourcesLoader {
     return loadedTextures[resourceName];
   }
 
+  /**
+   * Return the PIXI texture relative to a file.
+   * If not loaded, it will load it.
+   * @returns The PIXI.Texture to be used. It can be loading, so you
+   * should listen to PIXI.Texture `update` event, and refresh your object
+   * if this event is triggered.
+   *
+   * If the texture image resource is missing, GDevelop will try to add it
+   */
+  static getPixiTextureRelativeToFile(
+    project: gdProject,
+    resourceName: string,
+    relativeToFile: string
+  ) {
+    if (!project.getResourcesManager().hasResource(resourceName)) {
+      const projectPath = path.dirname(project.getProjectFile());
+      const fullPathRelativeFile = path.dirname(
+        path.join(
+          projectPath,
+          project
+            .getResourcesManager()
+            .getResource(relativeToFile)
+            .getFile()
+        )
+      );
+      const fullPathNewResource = path.resolve(
+        fullPathRelativeFile,
+        resourceName
+      );
+      const resourceRelativePath = path.relative(
+        projectPath,
+        fullPathNewResource
+      );
+
+      const newResource = new gd.ImageResource();
+      newResource.setName(resourceName);
+      newResource.setFile(resourceRelativePath);
+      project.getResourcesManager().addResource(newResource);
```

Comment:
This work but is fairly hacky to dynamically add resources while loading one - we'll have to think about a way to handle this without adding resources on the fly at a moment where it would be supposed not to do this (but just return a pixi texture without modifying the project).

Reply:
do you think the engine itself needs the concept of resource dependencies? I can't think of another way to do it